# Prepare reference data used in APIARC.This is a help workflow to download all reference data for all workflows.

| File Name          | Position          | Meaning          | Workflow          | Tool          |  Link         |			|        
|:-----          |:-----           |:-----           |:-----           |:-----           |:------       |:-----			|			            
|sample_1_val_1.fq.gz(paired)           |QCfile           |-           |RNAProj/CHIPProj           |trim_galore           |https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/            |
|sample_2_val_2.fq.gz(paired)           |QCfile           |-           |RNAProj/CHIPProj           |trim_galore           |https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/            |
|sample_1_val_1_fastqc.html(paired)           |QCfile           |-           |RNAProj/CHIPProj           |trim_galore           |https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/	|
|sample_2_val_2_fastqc.html(paired)           |QCfile           |-           |RNAProj/CHIPProj           |trim_galore           |https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/	|
|sample_1_val_1_fastqc.zip(paired)           |QCfile           |-           |RNAProj/CHIPProj           |trim_galore           |https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/	|
|sample_2_val_2_fastqc.zip(paired)           |QCfile           |-           |RNAProj/CHIPProj           |trim_galore           |https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/	|
|sample_1.fastq_trimming_report.txt(paired)           |QCfile           |-           |RNAProj/CHIPProj           |trim_galore           |https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/	|
|sample_2.fastq_trimming_report.txt(paired)           |QCfile           |-           |RNAProj/CHIPProj           |trim_galore           |https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/	|
|sample_1_trimmed.fq.gz(single)           |QCfile           |-           |RNAProj/CHIPProj          |trim_galore           |https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/	|
|sample_1_trimmed_fastqc.html(single)           |QCfile           |-           |RNAProj/CHIPProj           |trim_galore           |https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/	|
|sample_1_trimmed_fastqc.zip(single)           |QCfile           |-           |RNAProj/CHIPProj           |trim_galore           |https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/	|
|sample_1.fastq_trimming_report.txt(single)           |QCfile           |-           |RNAProj/CHIPProj           |trim_galore           |https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/	|
|sample_clean_1.fq.gz(paired)           |QCfile/fastp/paired  |    -       |  RNAProj         | fastp          |https://github.com/OpenGene/fastp	|
|sample_clean_2.fq.gz(paired)           |QCfile/fastp/paired  |      -    |     RNAProj      |fstap           |	https://github.com/OpenGene/fastp|
|sample.json(paired)           |QCfile/fastp/paired  |  -         |      RNAProj     | fstap          |	https://github.com/OpenGene/fastp|
|sample.html(paired)           |QCfile/fastp/paired  |   -        |     RNAProj      |  fastp         |https://github.com/OpenGene/fastp	|
|sample_clean.fq.gz(single)           |QCfile/fastp/single  |    -       |  RNAProj         | fastp          |https://github.com/OpenGene/fastp	|
|sample.json(single)           |QCfile/fastp/single  |  -         |      RNAProj     | fstap          |	https://github.com/OpenGene/fastp|
|sample.html(single)           |QCfile/fastp/single  |   -        |     RNAProj      |  fastp         |https://github.com/OpenGene/fastp	|
|sample.sam           | mapping          |   -        |  RNAProj         |  hisat2         |https://open.bioqueue.org/home/knowledge/showKnowledge/sig/hisat2	|
|hisat2_summary.log           |    mapping       | It contains the alignment logs of each sample generated by HISAT2.          |     RNAProj      |       -    |-	|
| sample_filtered.sam          |  mapping         |    It contains filtered SAM files.       |    RNAProj       |       -   |	-|
|sample_sorted.bam           |    mapping       |  -         |    RNAProj       |  samtools         |https://www.htslib.org	|
|sample_sorted.bam.bai           |   mapping        |   -        |      RNAProj     |    samtools       |https://www.htslib.org	|
|sample.dedup.bam           |mapping           |     -      |     RNAProj      |   picard        |	https://cloud.tencent.com/developer/article/2480345|
|sample.metricsFile           |  mapping         |     -      |     RNAProj      |    picard       |	https://cloud.tencent.com/developer/article/2480345|
|sample.dedup.bam.bai           |  mapping         |  -         |    RNAProj       |   samtools        |https://www.htslib.org	|
|sample.transcripts.gtf           |   expression        |      -     |    RNAProj       |   stringtie        |	https://ccb.jhu.edu/software/stringtie/index.shtml?t=manual|
|sample.gene_abundance.txt           |   expression        |     -      |   RNAProj        |  stringtie         |	https://ccb.jhu.edu/software/stringtie/index.shtml?t=manual|
|sample_list.txt           |   expression        |     It contains the paths of each GTF fileS and their corresponding names( you refer to in the Config.)      |     RNAProj      |      -     |	-|
|transcript_count_matrix.csv           |   DEG        | The original count matrix at the transcript level.          |    RNAProj       |      -     |	-|
|gene_count_matrix.csv           |     DEG      |  The original count matrix at the gene level.         |    RNAProj      |  -        |	-|
|result.csv           |  DEG         |   -        |    RNAProj      |           |	https://bioconductor.org/packages/release/bioc/html/DESeq2.html|
|UP_gene_names.bed           |   DEG        |    -       |   RNAProj       |           |https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html	|
| UP_genes_name.csv          |DEG           |    -       |  RNAProj        |           |	https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html|
|DOWN_gene_names.bed           |  DEG         |  -         |    RNAProj       |           |https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html	|
| DOWN_gene_names.csv          |DEG          |   -        | RNAProj          |           |	https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html|
|hista2_result.pdf           |     picture      |    -       |  RNAProj           |           |	|
| mRNA_volcano.pdf          | picture          |   -        |      RNAProj       |           |	|
| sample_boxplot.pdf          |   picture        |   -        |     RNAProj        |           |	|
|sample_dist_heatmap.pdf           |  picture         |   -        | RNAProj            |           |	|
| sample_PCA.pdf          |  picture         |     -      |  RNAProj           |             |	|
|           |           |           |           |           |	|








